lib/zig.h

```diff
@@ -180,10 +180,16 @@ typedef char bool;
 #define zig_export(sig, symbol, name) __asm(name " = " symbol)
 #endif
 
+#if zig_has_builtin(trap)
+#define zig_trap() __builtin_trap()
+#elif defined(__i386__) || defined(__x86_64__)
+#define zig_trap() __asm__ volatile("ud2");
+#else
+#define zig_trap() abort() // This is what LLVM does (https://llvm.org/docs/LangRef.html#id1528); TODO: should we `raise(SIGILL)` instead?
```

Comment:
I initially went with abort here because that's what LLVM does and also `__builtin_trap`: https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html

>  GCC implements this function by using a target-dependent mechanism (such as intentionally executing an illegal instruction) or by calling `abort`

Note how for `debugtrap` we fall back to `raise(SIGTRAP)`. That's why I suggested `raise(SIGILL)` for this.

SIGTRAP:
> The SIGTRAP signal is sent to a process when an exception (or trap) occurs: a condition that a debugger has requested to be informed of – for example, when a particular function is executed, or when a particular variable changes value.

SIGILL:
> The SIGILL signal is sent to a process when it attempts to execute an illegal, malformed, unknown, or privileged instruction.

Both signals seem perfect for each case.

Reply:
Went with `raise(SIGILL)` for now.